chore: add graphile job metrics to prom

xvello opened the pull request:
## Problem

Part of https://github.com/PostHog/posthog-cloud-infra/issues/1052, bring our alerting to prometheus.

## Changes

Add prometheus metrics around plugin-server job execution:

  - ~`jobs_enqueued_total` and `jobs_enqueue_failures_total` in the `createJobs` helper (writes to Kafka)~ dropped this for now, as that code runs in a piscina worker and cannot currently export prom metrics. We'll use the "message count to kafka jobs topic" metric for now
  - `jobs_consumer_enqueue_success_total` and `jobs_consumer_enqueue_failures_total` in the Kafka->Graphile jobs-consumer
  - `jobs_triggered_total` in the graphile worker (reads from Graphile queue and call plugins)

## How did you test this code?

#yolo

xvello committed:
chore: add graphile job metrics to prom

hazzadous commented on plugin-server/src/worker/vm/extensions/jobs.ts:
```diff
@@ -49,7 +67,9 @@ export function createJobs(server: Hub, pluginConfig: PluginConfig): Jobs {
             }
             server.statsd?.increment('job_enqueue_attempt')
             await server.enqueuePluginJob(job)
+            jobsEnqueuedCounter.inc()
```
I've got a feeling that this is run in Piscina and therefore you won't see the keys in `/_metrics` as it stands, or is that not the case?

xvello replied:
Arf, you are right!
Until we get the piscina worker stuff sorted out, could we use the number of kafka messages written to the `jobs` topic?

xvello replied:
Indeed, they stay at zero due to the multiprocess. `prom-client`'s multi-process support relies on the `cluster` lib and I'm not sure whether it would play nice with piscina.

Removing them and going with the kakfa metric for now.

hazzadous commented on plugin-server/src/main/graphile-worker/worker-setup.ts:
```diff
@@ -66,6 +72,7 @@ export function getPluginJobHandlers(hub: Hub, graphileWorker: GraphileWorker, p
     const pluginJobHandlers: TaskList = {
         pluginJob: async (job) => {
             pauseQueueIfWorkerFull(() => graphileWorker.pause(), hub, piscina)
+            jobsTriggeredCounter.inc()
```
non-blocking as it's an improvement rather than straight lift and shift

Should we add in a success/failure after we're executed?

xvello replied:
Good idea, will do!

hazzadous reviewed (approved):
Looks good, just a couple of questions that may or may not be relevant.

xvello committed:
add success/failure metric + job_type label

xvello committed:
remove job emission metric

They are collected at zero, because this code runs in the piscina
workers. For now, we'll use the kakfa message in metric.
